src/app/static/src/app/apiService.ts

```diff
@@ -74,19 +74,20 @@ export class APIService<S extends string, E extends string> implements API<S, E>
     };
 
     withSuccess = (type: S) => {
-        this._onSuccess = (success: Response) => {
-            this._commit({type: type, payload: freezer.deepFreeze(success.data)});
+        this._onSuccess = (data: any) => {
+            this._commit({type: type, payload: freezer.deepFreeze(data)});
         };
         return this;
     };
 
     private _handleAxiosResponse(promise: Promise<AxiosResponse>) {
         return promise.then((response: AxiosResponse) => {
             const success = response && response.data;
+            const data = success.data;
             if (this._onSuccess) {
-                this._onSuccess(success);
+                this._onSuccess(data);
             }
-            return success;
+            return data;
```

Comment:
Figured it's actually much more useful to return the response data rather than the whole response object. We weren't actually using this anywhere yet so this change hasn't affected any other code.